### ISSUE TITLE
Process input path variable contains wrong data type

### ISSUE THREAD

**Issue by marcadella** [nextflow-io/nextflow#1653]
## Bug report 

Given a process with the following input:
```
input:
    path input, stageAs: 'in/*'
```
where the input channel is a list of paths (`Channel.fromPath(params.reads).toSortedList()`), I expect that `input[0]` will return the first path (staged under `in`), `input[1]` will return the second path and so on. This is what happens if the input channel contains at least two paths.  
For instance, given the two files `myFile1.txt` and `myFile2.txt` (piping the channel into `view` prints: `[myFile1.txt, myFile2.txt]`), if echo the array `input`, I get:
- `input[0]` = "in/myFile1.txt"
- `input[1]` = "in/myFile.txt"
- `input[2]` = "null"

However if the input channel contains only one path called `myFile.txt` (piping the channel into `view` prints: `[myFile.txt]`) here is what I get when I echo the `input` array:
- `input[0]` = "in"
- `input[1]` = "myFile.txt"
- `input[2]` = "null"

while I expected:
- `input[0]` = "in/myFile.txt"
- `input[1]` = "null"
- `input[2]` = "null"

### Environment 

* Nextflow version: 20.04.1.5335
* Java version: 1.8.0_252
* Operating system: Linux
* Bash version: GNU bash, version 4.4.20(1)-release (x86_64-pc-linux-gnu)

**Comment by pditommaso** [comment]
Please include a replicable test case.

**Comment by marcadella** [comment]
# Test

We define a channel containing all the paths ending in `.txt`. We first print the 3 first paths using a `view` (prefixed with `>`), the we print them again but within a process (without `>`).

```
#!/usr/bin/env nextflow

nextflow.preview.dsl = 2

process Proc {
  echo true

  input:
    path input, stageAs: 'in/*'

  shell:
    '''
    echo !{input[0]}
    echo !{input[1]}
    echo !{input[2]}
    '''
}

workflow {
    file_ch = Channel.fromPath("*.txt").toSortedList() | view { array ->
    "> ${array[0]}\n> ${array[1]}\n> ${array[2]}"
    }
    Proc(file_ch)
} 
```

## Case where there is no `.txt` file in the directory here is the output
```
> null
> null
> null
null
null
null
```
which is what we expect.

## Case where there are 2 `.txt` files:
```
> /home/xxx/test/nextflowPaths/a.txt
> /home/xxx/test/nextflowPaths/b.txt
> null
in/a.txt
in/b.txt
null

```
which is what we expect.

## Case where there is 1 `.txt` file:
```
> /home/xxx/test/nextflowPaths/a.txt
> null
> null
in
a.txt
null
```

Here you can see that `input` contains two paths: `in` and `a.txt`, while I would have expected `in/a.txt` as only path.

**Comment by pditommaso** [comment]
bump

**Comment by bentsherman** [comment]
This issue happens because the list is unwrapped if there is only one file. You can fix it like so:

```groovy
process Proc {
  debug true

  input:
    path input, stageAs: 'in/*'

  shell:
    inputs = input instanceof Path ? [input] : input
    '''
    echo !{inputs[0]}
    echo !{inputs[1]}
    echo !{inputs[2]}
    '''
}
```

Also it will be fixed by #3706 , then you will be able to specify `arity: true` to keep the list from being unwrapped:

```groovy
process Proc {
  debug true

  input:
    path input, stageAs: 'in/*', arity: true

  shell:
    '''
    echo !{input[0]}
    echo !{input[1]}
    echo !{input[2]}
    '''
}
```

**Comment by bentsherman** [comment]
Fixed by https://github.com/nextflow-io/nextflow/commit/42504d3c83145e16179e22df83e9d35ecc995eca
